src/wire/tcp.rs

```diff
@@ -690,7 +692,19 @@ impl<'a> Repr<'a> {
                 TcpOption::NoOperation => (),
                 TcpOption::MaxSegmentSize(value) =>
                     max_seg_size = Some(value),
-                _ => ()
+                TcpOption::WindowScale(value) => {
+                    // RFC-1323: Thus, the shift count must be limited to 14 (which allows windows
+                    // of 2**30 = 1 Gbyte). If a Window Scale option is received with a shift.cnt
+                    // value exceeding 14, the TCP should log the error but use 14 instead of the
+                    // specified value.
+                    window_scale = if value > 14 {
+                        net_debug!("parsed window scaling factor >14, setting to 14");
```

Comment:
Take a look at how all other log statements are written--you're missing the socket endpoints in this one.

Reply:
Since this is packet parsing code that does not contain an obvious reference to a socket, should I just print the source and destination addresses and ports? That means the fields will be reversed between packets we send and packets we receive, but it would be better than nothing.

Reply:
Hmm, the obvious way looks a bit uglier than I thought:

```rust
net_debug!("{}->{}: parsed window scaling factor >14, setting to 14",
           IpEndpoint::new(src_addr.clone(), packet.src_port()), // borrowed src_addr, must be owned
           IpEndpoint::new(dst_addr.clone(), packet.dst_port()));  // borrowed dst_addr, must be owned
```

Reply:
Nevermind, I was being too literal-minded. I think I've got it.